listener/inventory_processor.py

```diff
@@ -175,6 +175,7 @@ def _parse_system_data(self, msg: InventoryMsg) -> (str, list):
     async def _db_import_repos(self, conn, repo_list: list) -> [str]:
         """Import not yet imported repos to db"""
         repo_names = []
+        repo_id_mapping = {}
         to_insert = sorted([(repo,) for repo in repo_list if repo not in self.repo_id_cache])
```

Comment:
one idea - if we copy requested repos from `self.repo_id_cache` (that are already cached) into `repo_id_mapping`, then we can avoid all new conditions checking presence in `_db_import_system_repos` and `_db_unlink_system_repos`